### ISSUE TITLE
Normals on Meshes? 

### ISSUE THREAD

**Issue by honeydewloon** [Roblox/avatar-evolution#8]
I notice that I can't seem to apply normal maps and other PBR channels to meshes that I have Unwrapped and baked myself. From what I have gathered, it seems like it only works on parts. Is this an issue or was it intended, and if it was intended, will we have working PBR settings for meshes as well?

**Comment by Rocky28447** [comment]
They work just fine for me.

![image](https://user-images.githubusercontent.com/26193486/70850608-38f9b400-1e5a-11ea-9d28-1645e46448bb.png)
![image](https://user-images.githubusercontent.com/26193486/70850639-8e35c580-1e5a-11ea-9baa-7d7313533ef8.png)

**Comment by honeydewloon** [comment]
Interesting, i'll give it another go and share my findings if I come across something I believe to be the problem. Thanks for the response brother!

**Comment by Rocky28447** [comment]
I've discovered that if you import the meshes using production studio, SurfaceAppearances will work just fine. If you import them in this build, they will not work.

**Comment by ferrarif340** [comment]
I've had this issue too. I was really frustrated but I'll try using production studio to fix it.

**Comment by ferrarif340** [comment]
Nope, still not working for me. I've tried most things and nothing has worked. I'm unsure as to what the problem is because, as honey described, it works on parts but not MeshParts.

**Comment by LaAerooo** [comment]
Same goes for me. All it does is remove the shading

**Comment by honeydewloon** [comment]
Yeah I really don't know, I have tried uploading with production and then adding SurfaceAppearence in Beta Studio. Nothing seems to work

**Comment by jbrosnihan-roblox** [comment]
If you are not seeing metal reflections, follow the instructions here: https://roblox.github.io/avatar-evolution/surface-appearance
Try setting Settings > Rendering > Edit Quality Level to 21. If that doesn't work, please post your OS + graphics card information and a .rbxl file of the place where you aren't seeing the reflections.

If you are not seeing normal maps show up on an imported mesh, make sure that you enabled the equivalent "Export Tangents" property in your 3D modeling software before exporting your mesh file. If that doesn't work, try importing the S15_Mechy.fbx file in the article above and applying the provided normal map on him. If that works but normal maps have no effect on your own mesh, that implies your mesh is missing tangent data. If you import Mechy and his normal map doesn't work, please post your OS + graphics card info and the .rbxl file containing the imported rig.

If you exported tangents with your mesh but still aren't seeing normal maps have any effect, please post your mesh file so I can debug it.

**Comment by honeydewloon** [comment]
The problem was that I needed to export using an OBJ file format. I may be wrong, but I didn't see any information clarifying that this was needed.

**Comment by Rocky28447** [comment]
Update update: Looks like it has nothing to do with what version you upload the mesh in. What matters is the format of the uploaded 3d mesh. In both versions of studio, uploading a **.fbx file** will cause normals to **not work**. In both versions of studio, uploading a **.obj file** will have the normals work just fine.

I'm unsure if there's a special setting you need to set when exporting as a .fbx to allow normals to work. I could not find any setting called "Export Tangents" as @jovannic said.

**Comment by honeydewloon** [comment]
@Rocky28447 Yeah, I eventually figured it out. Thanks for your help & patience. Again, I may be wrong in saying that ROBLOX did not clarify what format that the exported mesh needed to be in, but it would be really helpful if they included it in their guide.

**Comment by Rocky28447** [comment]
I'm sure this is not intended behavior, as you've historically been able to upload either fbx files or obj files.

**Comment by jbrosnihan-roblox** [comment]
OBJ and FBX should both work. @Rocky28447 What program are you exporting these from? If you can send me the OBJ and FBX, it would help me debug the issue. You can upload a zip file on this GH issue, email jbrosnihan@roblox.com, or send to Homeomorph on the dev forum.

**Comment by jbrosnihan-roblox** [comment]
If you haven't already, try importing your FBX in the latest release build. https://github.com/Roblox/avatar-evolution/releases
We have a bug fix in there that may help.

**Comment by Rocky28447** [comment]
I exported them from blender and imported them via the folder button on MeshPart. I believe I've tried importing in release #2 but I can try again just to make sure. If it does not work still I will send the mesh files.

**Comment by Rocky28447** [comment]
Still does not work, even in version 2. Perhaps I'm doing something wrong as I'm no master of 3D modelling programs, but I don't think I am. Here are the mesh files (obj and fbx):

[PBR_SurfaceAppearanceTest_Gauntlet.zip](https://github.com/Roblox/avatar-evolution/files/3982393/PBR_SurfaceAppearanceTest_Gauntlet.zip)

**Comment by Reshiram110** [comment]
I can confirm, `SurfanceAppearance` instances do not seem to affect imported meshes, specifically those with bone rigs in them.
